cleanlab/multilabel_classification.py

```diff
@@ -0,0 +1,112 @@
+    pred_probs: np.ndarray,
+    *,
+    method: str = "self_confidence",
+    adjust_pred_probs: bool = False,
+    aggregator_kwargs: dict = {"method": "exponential_moving_average", "alpha": 0.8}
+) -> np.ndarray:
+    """Returns a label quality score for each datapoint.
+
+    This is a function to compute label quality scores for multi-label classification datasets,
+    where lower scores indicate labels more likely to contain an error.
+
+    Score is between 0 and 1 with lower values indicating labels inferred to be worse.
+    For each example, this method internally computes a separate score for each individual class
+    and then aggregates these per-class scores into an overall label quality score for the example.
+
+    Parameters
+    ----------
+    labels : List[List[int]]
+      Multi-label classification labels for each example, which is allowed to belong to multiple classes.
+      The i-th element of `labels` corresponds to list of classes that i-th example belongs to (e.g. ``labels = [[1,2],[1],[0],..]``).
+      *Format requirements*: for dataset with K classes, individual class labels must be integers in 0, 1, ..., K-1.
+
+    pred_probs : np.ndarray
+      An array of shape ``(N, K)`` of model-predicted probabilities,
+      ``P(label=k|x)``. Each row of this matrix corresponds
+      to an example `x` and contains the model-predicted probabilities that
+      `x` belongs to each possible class, for each of the K classes. The
+      columns must be ordered such that these probabilities correspond to
+      class 0, 1, ..., K-1. In multi-label classification, the rows of `pred_probs` need not sum to 1.
+
+      **Note**: Estimated label quality scores are most accurate when they are computed based on out-of-sample `pred_probs` from your model.
+      To obtain out-of-sample predicted probabilities for every example in your dataset, you can use :ref:`cross-validation <pred_probs_cross_val>`.
+      This is encouraged to get better results.
+
+    method : {"self_confidence", "normalized_margin", "confidence_weighted_entropy"}, default = "self_confidence"
+      Method to calculate separate per class annotation scores that are subsequently aggregated to form an overall label quality score.
+      These scores are separately calculated for each class based on the corresponding column of `pred_probs` in a one-vs-rest manner,
+      and are standard label quality scores for multi-class classification.
+      See :py:func:`rank.get_label_quality_scores <cleanlab.rank.get_label_quality_scores>` function for details about each option.
+
+    adjust_pred_probs : bool, default = False
+      Account for class imbalance in the label-quality scoring by adjusting predicted probabilities
+      via subtraction of class confident thresholds and renormalization.
+      Set this to ``True`` if you prefer to account for class-imbalance.
+      See `Northcutt et al., 2021 <https://jair.org/index.php/jair/article/view/12125>`_.
```

Comment:
I don't think user needs to worry about such internals, the idea they should have is just that it will adjust the label quality scores based on class confident thresholds. This will effectively still be doing that for the overall score too after pooling.